### PR TITLE
Added k8sio-dns-update postsubmit k/k8s.io prowjob

### PR DESCRIPTION
**Pull request by bartsmykla**
ref. [`k/k8s.io#783`](https://github.com/kubernetes/k8s.io/issues/783)

cc. @spiffxp

### CODE REVIEW

File: config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml [hunk 1]
```diff
@@ -44,3 +44,24 @@ postsubmits:
...
+      testgrid-num-failures-to-alert: '1'
+    spec:
+      serviceAccountName: k8s-infra-dns-updater
+      containers:
+      - image: us.gcr.io/k8s-artifacts-prod/infra-tools/octodns:v20200612-d4db040
+        command:
+        - bash
+        args:
+        - -c
+        - "cd dns && ./push.sh --dry-run"
```
Comment (spiffxp) [inline config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml:67]: consider using `make` as your main entrypoint
```suggestion
        - "cd dns && make dry-run-local"
```

Comment (bartsmykla) [reply config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml:67]: I was thinking about it but currently there is no make present in the image

**Pull request merged**